File: cmd/models/team.go
package models

import "time"

type Team struct {
	ID             string       `json:"id"`
	Name           string       `json:"name"`
	OrganizationID string       `json:"organizationId"`
	CreatedAt      time.Time    `json:"createdAt"`
	Members        []TeamMember `json:"members,omitempty"`
}

type TeamMember struct {
	ID       string    `json:"id"`
	TeamID   string    `json:"teamId"`
	UserID   string    `json:"userId"`
	User     *User     `json:"user,omitempty"`
	Role     string    `json:"role"`
	JoinedAt time.Time `json:"joinedAt"`
}

File: app/src/types/team.ts
import type { User, UserRole } from './user'

export interface Organization {
  id: string
  name: string
  ownerId: string
  createdAt: string
}

export interface Team {
  id: string
  name: string
  organizationId: string
  createdAt: string
}

export interface TeamMember {
  id: string
  teamId: string
  userId: string
  user: User
  role: UserRole
  joinedAt: string
}

export interface TeamWithMembers extends Team {
  members: TeamMember[]
}
